### ISSUE TITLE
B2. iterate on gcis:accessURL

### ISSUE THREAD

**Issue by zednis** [USGCRP/gcis-ontology#22]
from the GCIS Ontology Proposed Changes google doc:

> B2. Slightly confusing and possibly contradictory wording within definition for http://data.globalchange.gov/gcis.owl#accessURL
> Don’t understand the inclusion of word “download” in the definition for gcis:accessURL given the existence of gcis:downloadURL.  Therefore, clarify the distinction.

from current gcis.ttl:

```
gcis:accessURL a owl:DatatypeProperty ;
    rdfs:label "Access URL" ;
    rdfs:comment "Could be any kind of URL that gives access to a distribution of the dataset. E.g. landing page, download, feed URL, SPARQL endpoint. Use when your catalog does not have information on which it is or when it is definitely not a download." ;
    rdfs:domain gcis:Dataset ;
    rdfs:range xsd:anyURI .

gcis:downloadURL a owl:DatatypeProperty ;
    rdfs:label "Download URL" ;
    rdfs:comment "This is a direct link to a downloadable file in a given format. E.g. CSV file or RDF file. The format is described by the distribution's dc:format and/or dcat:mediaType." ;
    rdfs:domain gcis:Dataset ;
    rdfs:range xsd:anyURI .
```

From my reading of the properties I think the author was trying to replicate the definitions of the similarly named `dcat:accessURL` and `dcat:downloadURL` properties.  (see http://www.w3.org/TR/vocab-dcat/#Property:distribution_accessurl).

My guess is the intent is that `gcis:accessURL` be used when the URL references a landing page or service that provides access to the dataset but is not a URL of the dataset download itself.

Another way to put it - if the property used is `dcat:downloadURL` then I should be able to curl the URL and get back the dataset file.  If the property used is `dcat:accessURL` then I may have to go to a website page or interact with a webservice to get the file.

**Comment by xgmachina** [comment]
hi @zednis Yes I adapted the definitions from DCAT. Did not reuse the DCAT properties directly because the assertions of domain and range.

**Comment by zednis** [comment]
In the interest of compatibility and reuse of widely known vocabularies, would it make sense for us to adopt `dcat:accessURL` and `dcat:downloadURL`?  If we do this we would have to update the templates to include distributions for datasets.

**Comment by justgo129** [comment]
I like this idea.  Team, given that dcat:accessURL and dcat:downloadURL have domain dcat:Dataset (http://www.w3.org/TR/vocab-dcat/#Class:_Dataset), can we confirm that every dataset in GCIS qualifies as a dcat:Dataset, thereby allowing use of these properties?  Would we need to redefine gcis:Dataset a subclass (or owl:sameAs) as a dcat:Dataset?  Would that be problematic for the other dataset properties in the database?
https://data.globalchange.gov/gcis.owl#d4e1684

**Comment by zednis** [comment]
@justgo129 `dcat:accessURL` and `dcat:downloadURL` have domain `dcat:Distribution`.  Also, the DCAT properties are object properties whereas the GCIS properties are datatype properties.

current:

```
<http://data.globalchange.gov/dataset/nasa-nsidcdaac-0001>
  a gcis:Dataset ;
  gcis:accessURL "http://nsidc.org/data/NSIDC-0001"^^xsd:anyURI .
```

with DCAT:

```
<http://data.globalchange.gov/dataset/nasa-nsidcdaac-0001>
  a gcis:Dataset, dcat:Dataset ;
  dcat:distribution [
    a dcat:Distribution ;
    dcat:accessURL <http://nsidc.org/data/NSIDC-0001> ;
 ] .
```

The DCAT model allows a dataset to have multiple distributions (serializations) which can each have different access or download URLs.

**Comment by zednis** [comment]
DCAT does include a `dcat:landingPage` property which has domain `dcat:Dataset` as a more general dataset-level relationship.

definition: _A Web page that can be navigated to in a Web browser to gain access to the dataset, its distributions and/or additional information._

```
<http://data.globalchange.gov/dataset/nasa-nsidcdaac-0001>
  a gcis:Dataset, dcat:Dataset ;
  dcat:landingPage <http://nsidc.org/data/NSIDC-0001> .
```

This might make more sense for GCIS accessURLs since they were (presumably) collected at the dataset-level.

**Comment by justgo129** [comment]
@aulenbac @rewolfe what do you think?

**Comment by rewolfe** [comment]
I think this is fine for a landing page url.

On Wed, Jul 1, 2015 at 1:02 PM, justgo129 notifications@github.com wrote:

> @aulenbac https://github.com/aulenbac @rewolfe
> https://github.com/rewolfe what do you think?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/issues/22#issuecomment-117751945
> .

## 

Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

**Comment by bduggan** [comment]
On Wednesday, July  1, Stephan Zednik wrote: 

> DCAT does include a `dcat:hasLandingPage` property which has domain `dcat:Dataset` as a more general dataset-level relationship.

:+1:

The URLs we store are landing pages, not access URLs, so this is more appriopriate.

i.e. providers often have a number of URLs for dataset, and we try to use the
landing page, if there is one.

Brian

**Comment by zednis** [comment]
Thanks Brian, Robert,

My recommendation then is to change the RDF templates to use `dcat:landingPage`.

I also think we should consider removing `gcis:accessURL` and `gcis:downloadURL`.

**Comment by aulenbac** [comment]
As @bduggan mentions, we try to use landing pages if there is one. This is in keeping with best practices and recommendations from a GCIS advisory group. I agree with @bduggan and @rewolfe.

**Comment by zednis** [comment]
Have we reached a consensus on this issue?

**Comment by bduggan** [comment]
Yes, I think so.

**Comment by justgo129** [comment]
Fine here.  Within GCIS, are there any instances of the non-landing page URL being provided in-lieu of that for a landing page?  @aulenbac and @bduggan are right in that we try to use landing page - if there is one.

**Comment by bduggan** [comment]
On Wednesday, July 15, justgo129 wrote: 

> Fine here.  Within GCIS, are there any instances of the non-landing page
> URL being provided in-lieu of that for a landing page?

Maybe, but this should not inform the semantic model.

Brian

**Comment by zednis** [comment]
I will create a pull request to update the templates to use `dcat:landingPage`.

**Comment by zednis** [comment]
Should we remove `gcis:accessURL` and `gcis:downloadUrl` from the ontology?

**Comment by justgo129** [comment]
+1 for removing gcis:accessURL and gcis:downloadUrl from the ontology.

**Comment by justgo129** [comment]
I can prepare the pull request for removing the two aforementioned items and close this ticket.  @rewolfe is this solution acceptable?

**Comment by justgo129** [comment]
Closed #22 via merged #109.
